File: backend/models/post.go
// Autores:
//   - Jennyfer Jasso

package models

// Estructura de la tabla de posts
type Post struct {
	ID      uint   `json:"id" gorm:"primarykey" validate:"omitempty,eq=0"`
	Title   string `json:"title" validate:"required,min=1"`
	Author  string `json:"author" validate:"required,min=1"`
	Date    string `json:"date" validate:"required,min=1"`
	Content string `json:"content" validate:"required,min=1"`
}

type PostMetadata struct {
	Title   string `json:"title" validate:"required,min=1"`
	Author  string `json:"author" validate:"required,min=1"`
	Date    string `json:"date" validate:"required,min=1"`
	Content string `json:"content" validate:"required,min=1"`
}

File: platform/src/utils/api/post.ts
import {get, del, post } from "@/utils/methods";

export interface Posts {
    id: number;
    title: string;
    author: string;
    date: string;
    content: string
}


   


export default {
    posts: {
        setPost: (
            posts: Posts
        ): Promise<Posts> => {
            return post("/posts", posts)
        },

        getPosts: (): Promise<Posts[]> =>{
            return get("/posts")
        },

        deletePost: (id: number): Promise<void> => {
            return del(`/posts/${id}`)
        },

        updatePost: (
            id: number,
            posts: Posts) : Promise<void> => {
            return post(`/posts/${id}`, posts)
        }



    }
}